monkey/common/event_queue/i_event_queue.py

```diff
@@ -0,0 +1,55 @@
+from abc import ABC, abstractmethod
+from typing import Any, Callable
+
+from common.events import AbstractEvent
+
+
+class IEventQueue(ABC):
+    """
+    Manages subscription and publishing of events
+    """
+
+    @abstractmethod
+    def subscribe_all(self, subscriber: Callable[[AbstractEvent], None]):
+        """
+        Subscribes a subscriber to all events
+
+        :param subscriber: Callable that should subscribe to events
+        """
+
+        pass
+
+    @abstractmethod
+    def subscribe_type(
+        self, event_type: AbstractEvent, subscriber: Callable[[AbstractEvent], None]
+    ):
+        """
+        Subscribes a subscriber to the specifed event type
+
+        :param event_type: Event type to which the subscriber should subscribe
+        :param subscriber: Callable that should subscribe to events
+        """
+
+        pass
+
+    @abstractmethod
+    def subscribe_tag(self, tag: str, subscriber: Callable[[AbstractEvent], None]):
+        """
+        Subscribes a subscriber to the specified event tag
+
+        :param tag: Event tag to which the subscriber should subscribe
+        :param subscriber: Callable that should subscribe to events
+        """
+
+        pass
+
+    @abstractmethod
+    def publish(self, event: AbstractEvent, data: Any):
```

Comment:
```suggestion
    def publish(self, event: AbstractEvent):
```

The event is the data.